common/codegen/src/writes/arithmetics_type.rs

```diff
@@ -45,24 +47,33 @@ pub fn codegen_arithmetic_type() {
 // limitations under the License.
 
 // This code is generated by common/codegen. DO NOT EDIT.
-use crate::DFPrimitiveType;
+use crate::Scalar;
 
 pub trait ResultTypeOfBinary {{
-    type AddMul: DFPrimitiveType;
-    type Minus: DFPrimitiveType;
-    type IntDiv: DFPrimitiveType;
-    type Modulo: DFPrimitiveType;
-    type LeastSuper: DFPrimitiveType;
+    type AddMul: Scalar;
```

Comment:
Scalar includes type `PrimitiveTypes` & `String` & `Boolean`, maybe we just need to make it as `PrimitiveType` ?

Reply:
> Scalar includes type `PrimitiveTypes` & `String` & `Boolean`, maybe we just need to make it as `PrimitiveType` ?

Yes, changed.